api/pkg/apis/v1alpha1/model/campaigncontainer.go

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ * SPDX-License-Identifier: MIT
+ */
+
+package model
+
+import (
+	"errors"
+)
+
+type CampaignContainerState struct {
+	ObjectMeta ObjectMeta               `json:"metadata,omitempty"`
+	Spec       *CampaignContainerSpec   `json:"spec,omitempty"`
+	Status     *CampaignContainerStatus `json:"status,omitempty"`
+}
+
+type CampaignContainerSpec struct {
+}
+
+type CampaignContainerStatus struct {
+	Properties map[string]string `json:"properties"`
+}
+
+func (c CampaignContainerSpec) DeepEquals(other IDeepEquals) (bool, error) {
+	return true, nil
+}
+
+func (c CampaignContainerState) DeepEquals(other IDeepEquals) (bool, error) {
+	otherC, ok := other.(CampaignContainerState)
+	if !ok {
+		return false, errors.New("parameter is not a CampaignContainerState type")
+	}
+
+	equal, err := c.ObjectMeta.DeepEquals(otherC.ObjectMeta)
+	if err != nil || !equal {
+		return equal, err
+	}
+
+	equal, err = c.Spec.DeepEquals(*otherC.Spec)
+	if err != nil || !equal {
+		return equal, err
+	}
+
```

Comment:
I notice all the deepEquals doesn't compare Status not only in your PR but also in the existing code base. Do you know why?

Reply:
deepEquals is often used to detect if an object has changed. A user submitted object usually doesn't have status associated with it, while the object is retrieved from the system has status. Hence status comparison is generally skipped.